collector/infiniband_linux.go

```diff
@@ -144,6 +145,11 @@ func infinibandPorts(infinibandPath, device string) ([]string, error) {
 func readMetric(directory, metricFile string) (uint64, error) {
 	metric, err := readUintFromFile(filepath.Join(directory, metricFile))
 	if err != nil {
+		// Handle case when counters are N/A (not available).
```

Comment:
Can you add a short comment summing up https://github.com/prometheus/node_exporter/pull/974#issuecomment-419659160 so we know that we can drop this workaround eventually?

Reply:
@discordianfish done